streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableJoinGraphNode.java

```diff
@@ -0,0 +1,195 @@
+        String thisKTableName;
+
+        private Builder() {
+        }
+
+        public static Builder aKTableJoinGraphNode() {
+            return new Builder();
+        }
+
+        public Builder withJoinThisProcessor(ProcessorSupplier joinThisProcessor) {
+            this.joinThisProcessor = joinThisProcessor;
+            return this;
+        }
+
+        public Builder withJoinOtherProcessor(ProcessorSupplier joinOtherProcessor) {
+            this.joinOtherProcessor = joinOtherProcessor;
+            return this;
+        }
+
+        public Builder withJoinMergeProcessor(ProcessorSupplier joinMergeProcessor) {
+            this.joinMergeProcessor = joinMergeProcessor;
+            return this;
+        }
+
+        public Builder withJoinThisName(String joinThisName) {
+            this.joinThisName = joinThisName;
+            return this;
+        }
+
+        public Builder withJoinOtherName(String joinOtherName) {
+            this.joinOtherName = joinOtherName;
+            return this;
+        }
+
+        public Builder withJoinMerggeName(String joinMerggeName) {
+            this.joinMerggeName = joinMerggeName;
+            return this;
+        }
+
+        public Builder withJoinThisStoreNames(String[] joinThisStoreNames) {
+            this.joinThisStoreNames = joinThisStoreNames;
+            return this;
+        }
+
+        public Builder withJoinOtherStoreNames(String[] joinOtherStoreNames) {
+            this.joinOtherStoreNames = joinOtherStoreNames;
+            return this;
+        }
+
+        public Builder withTopologyNodeType(TopologyNodeType topologyTopologyNodeType) {
+            this.topologyTopologyNodeType = topologyTopologyNodeType;
+            return this;
+        }
+
+        public Builder withName(String name) {
+            this.name = name;
+            return this;
+        }
+
+        public Builder withProcessDetails(ProcessDetails processDetails) {
+            this.processDetails = processDetails;
+            return this;
+        }
+
+        public Builder withOtherKtableName(String otherKTableName) {
+            this.otherKTableName = otherKTableName;
+            return this;
+        }
+
+        public Builder withThisKTableName(String thisKTableName) {
+            this.thisKTableName = thisKTableName;
+            return this;
+        }
+
+
+
+        public KTableJoinGraphNode build() {
+            KTableJoinGraphNode
+                kTableJoinGraphNode =
+                new KTableJoinGraphNode(topologyTopologyNodeType, joinThisProcessor, joinOtherProcessor, joinMergeProcessor, joinThisName, joinOtherName, joinMerggeName,
+                                        joinThisStoreNames, joinOtherStoreNames, otherKTableName, thisKTableName);
+            kTableJoinGraphNode.setName(name);
+            kTableJoinGraphNode.setProcessed(processDetails);
+            return kTableJoinGraphNode;
+        }
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Is it necessary to override equals and hashcode here? I didn't find a usage of them.